### DISCUSSION TITLE
How to add compiler flags for additional includes or libraries?

### DISCUSSION THREAD

**Discussion by friedrichgs** [idaholab/moose#20048] [Q&A General] [answered]
Hello,
how do I add compiler flags to the Makefile of an application?

I tried
`ADDITIONAL_CPPFLAGS = -I/path/to/include`
at the beginning of the application's Makefile, but it doesn't have an effect.

Besides additional includes, I'll also need to link additional libraries.

(A related question: Is it possible to set something like a VERBOSE flag to the make command, so that every executed detailed step during the build process will be shown?)

Thanks
Friedrich

**Comment by GiudGiud** [comment, marked as answer]
Hello

`ADDITIONAL_CPPFLAGS := -I/path/to/include` should be the way to do this. 
Use `make -n` to get a dry run with all the commands that would be run.
Or `make SHELL='sh -x'` to run the commands and get output. 

Guillaume

**Comment by friedrichgs** [comment]
Thanks, this is very helpful, problem solved.

https://github.com/idaholab/moose/discussions/19413 helped, too.

(I had a number of mistakes. Maybe this helps somebody else, who is not a Makefile expert, either:
1. I used ':=' instead of '=' (although '=' seems to work)
2. I wrote `ADDITIONAL_CPPFLAGS = -I$(shell pkg-config --cflags ...)` (Didn't show this in such detail above.) This was one '-I' too much, because the pkg-config output contains the -I already.

I also tried ADDITIONAL_INCLUDES instead of ADDITIONAL_CPPFLAGS. This seems to require '+=' instead of ':=', and it needs to appear before `include $(FRAMEWORK_DIR)/app.mk`, whereas `ADDITIONAL_CPPFLAGS :=...` can go anywhere.

So this one worked for me as well:
```
...
ADDITIONAL_INCLUDES += $(shell pkg-config --cflags ...)
include $(FRAMEWORK_DIR)/app.mk
...
```
)
